stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -985,3 +985,119 @@ func.func @concat_bounds_unranked_c1(
   %1 = "hlo_test_infer.get_return_types"(%result) : (tensor<5x?xi32>) -> tensor<*xindex>
   func.return %1 : tensor<*xindex>
 }
+
+// -----
+
+// CHECK-LABEL: @gather
+func.func @gather(%operand : tensor<*xi32>, %start_indices : tensor<1x5x2xi32>) -> tensor<3xi32> {
+  %result = "stablehlo.gather"(%operand, %start_indices) {
+    dimension_numbers = #stablehlo.gather<
+      collapsed_slice_dims = [0, 1],
+      index_vector_dim = 2,
+      offset_dims = [2],
+      start_index_map = [0, 1]
+    >,
+    indices_are_sorted = false,
+    slice_sizes = dense<[1, 1, 8]> : tensor<3xi64>
+  } : (tensor<*xi32>, tensor<1x5x2xi32>) -> tensor<1x5x8xi32>
+  %1 = "hlo_test_infer.reify_return_type_shapes"(%result) : (tensor<1x5x8xi32>) -> tensor<3xi32>
```

Comment:
It looks like this file has conventions for checking the types/shapes:
https://github.com/openxla/stablehlo/blob/9a224f1bd78ec6b68d6a0ad16e6c508f62627aa9/stablehlo/tests/infer_stablehlo.mlir#L6

Can these tests have a similar `CHECK:`